src/Util/Regex.php

```diff
@@ -33,7 +33,7 @@ public function __construct(string $string)
     private function isRegex(string $string): bool
     {
         if (preg_match('/^(.{3,}?)['.self::ALLOWED_MODIFIERS.']*$/', $string, $m)) {
-            $start = substr($m[1], 0, 1);
+            $start = $m[1][0];
```

Comment:
I would prefer to use a string function here to avoid confusion.

Reply:
Done